cpu/sam0_common/periph/Kconfig

```diff
@@ -5,5 +5,5 @@
 # directory for more details.
 
 config WDT_WARNING_PERIOD
-    depends on HAS_WDT_WARNING_PERIOD && KCONFIG_MODULE_PERIPH_WDT
+    depends on WDT_WARNING_PERIOD_SUPPORT && KCONFIG_MODULE_PERIPH_WDT
```

Comment:
Wouldn't it be easier for the future specify a list of well known prefixes? E.g. `HAS_` apparently is already one of these prefixes. And if `HAS_` does not apply here, I suggest to specify a second prefix. It would be needed to document this, so that the prefixes are used correctly.

Reply:
> Wouldn't it be easier for the future specify a list of well known prefixes? 
Good idea. We can add documentation on this.

Any suggestion for the prefix? `PROVIDES_`? `SUPPORTS_`?

Reply:
Both sound good to me. Maybe it could be defined that a verb as first word in a symbol is reserved for well known prefixes? All other Kconfig symbols present so far seem to be something like `CPU_...`, `APP_...`, `<MODULE_NAME>_...`. If I didn't overlook anything, the only `<verb>_` is `HAS_` so far.

My gut feeling is that "regular" symbol names would never look like `<verb>_...` anyway. And with that reservation in place, developers would recognize a new well known prefix as such even if they see it for the very first time.

Reply:
Indeed `HAS_` is the only `<verb>_` so far, but there are other well-known prefixes already:
- `MODULE_`: generated symbol from `USEMODULES`, indicates that a module is being used
- `PKG_`: generated symbol from `USEPKG`, indicates that a package is being used
- `KCONFIG_MODULE_`: Indicates that the module is being configured via Kconfig
- `KCONFIG_PKG_`: Indicates that the package is being configured via Kconfig

Reply:
Changed prefix to `PROVIDES_`.

Reply:
> Indeed `HAS_` is the only `<verb>_` so far, but there are other well-known prefixes already

It might still be sensible to reserve `<verb>_` for prefixes; that reservation would not imply that prefixes are limited to this.

(It would however likely be *feasible* to limit prefixes to a `<verb>_` pattern. E.g. `MODULE_` could be replaced by `USES_MODULE_`, `PKG_` to `USES_PKG_`. But *feasible* does not imply it is *sensible* :-))